app/controllers/homepage_controller.rb

```diff
@@ -0,0 +1,23 @@
+require 'rqrcode'
+
+class HomepageController < ApplicationController
```

Comment:
I think it would be cleaner to have the QR code functionality built into the Users controller `show` action.